apps/contentful-apps/infra/contentful-apps.ts

```diff
@@ -0,0 +1,28 @@
+import { service, ServiceBuilder } from '../../../infra/src/dsl/dsl'
+
+export const serviceSetup = (): ServiceBuilder<'contentful-apps'> =>
+  service('contentful-apps')
+    .image('contentful-apps')
+    .namespace('contentful-apps')
+    .serviceAccount('contentful-apps')
+    .ingress({
+      primary: {
+        host: {
+          dev: 'contentful-apps',
+          staging: 'contentful-apps',
+          prod: 'contentful-apps.devland.is',
+        },
```

Comment:
Note @island-is/devops. This app is to expand the UI in our CMS for developers and content editors. We don't need dev/staging/prod environments and we'd generally like to release updates outside the normal release trains.

So we'll probably deploy this like our existing "contentful-extensions-translations" project. Only host it on dev with CD, without cognito (nothing sensitive here) and point cognito to `https://contentful-apps.dev01.devland.is/something`.

My only concern is downtime during deployment. The existing project is static but this one is nextjs. Not sure if that's worse or if it's a big problem at all.

 ideally we can have zero-downtime continuous delivery for this app. It's used by developers and content editors

Reply:
> Note @island-is/devops. This app is to expand the UI in our CMS for developers and content editors. We don't need dev/staging/prod environments and we'd generally like to release updates outside the normal release trains.
> 
> So we'll probably deploy this like our existing "contentful-extensions-translations" project. Only host it on dev with CD, without cognito (nothing sensitive here) and point cognito to `https://contentful-apps.dev01.devland.is/something`.
> 
> My only concern is downtime during deployment. The existing project is static but this one is nextjs. Not sure if that's worse or if it's a big problem at all.
> 
> ideally we can have zero-downtime continuous delivery for this app. It's used by developers and content editors



regarding the "downtime during" deployment issue, are you referring to the indexer issue?

Reply:
We can't point cognito to that url, but we can disable authentication on the ingress for this service

Reply:
I finally got the TypeScript stuff to work. Can someone from @island-is/devops review this PR?